debezium-core/src/main/java/io/debezium/pipeline/ChangeEventSourceCoordinator.java

```diff
@@ -83,6 +88,9 @@ public ChangeEventSourceCoordinator(Offsets<P, O> previousOffsets, ErrorHandler
         this.executor = Threads.newSingleThreadExecutor(connectorType, connectorConfig.getLogicalName(), "change-event-source-coordinator");
         this.eventDispatcher = eventDispatcher;
         this.schema = schema;
+
+        notificationService = NotificationService.getInstance();
```

Comment:
I think we should pass the `NotificationService` in the same way as we pass `SignalProcessor` and all the initialization should be done at connector startup so we can use it at an arbitrary place.

Reply:
I was in doubt between the two approach until the end. But definitely maybe the other one is better. A Dependency Injection framework would be useful. 

I'll do the changes.

Reply:
Yes, DI would be useful :-(. Generally there is nothing wrong with your approach but I'd prefer to keep the code consistent with other services too.